### PR TITLE
refactor: use threading for concurrent requests

### PR DESCRIPTION
**Pull request by renaudjester**
Gets rid of aiohttp and nest-async

fix: [CMT-88](https://cms-change.atlassian.net/browse/CMT-88), [CMT-73](https://cms-change.atlassian.net/browse/CMT-73), [CMT-94](https://cms-change.atlassian.net/browse/CMT-94)

### CODE REVIEW

File: copernicusmarine/catalogue_parser/catalogue_parser.py [hunk 1]
```diff
@@ -49,87 +41,63 @@
...
-        if dataset_id in dataset_metadata_link.href
-    ]
-    if not datasets_metadata_links:
-        return None
-    # TODO: check if we gain a lot of time by doing a gather here
-    # if not don't forget to add a retry policy
-
-    dataset_jsons: list[dict] = await asyncio.gather(
-        *[
+    with CatalogParserConnection() as connection:
```
Comment (uriii3) [inline copernicusmarine/catalogue_parser/catalogue_parser.py:68]: this also helps to solve some of the warnings in [here](https://github.com/Deltares/dfm_tools/issues/803), no?

Comment (renaudjester) [reply copernicusmarine/catalogue_parser/catalogue_parser.py:68]: Maybe, let's check the warnings when we merged everything

File: copernicusmarine/download_functions/download_original_files.py [hunk 2]
```diff
@@ -285,11 +257,11 @@ def _download_header(
     only_list_root_path: bool = False,
     overwrite: bool = False,
 ) -> Optional[
-    Tuple[
-        Tuple[str, str],
+    tuple[
+        tuple[str, str],
```
Comment (uriii3) [inline copernicusmarine/download_functions/download_original_files.py:261]: so we are now using the default version of tuple from python?

File: copernicusmarine/download_functions/download_original_files.py [hunk 3]
```diff
@@ -539,61 +511,43 @@ def _get_file_size_and_last_modified(
...
 
 
-def _download_files(
-    tuple_original_files_filename: Tuple[
-        str, str, str, list[str], list[pathlib.Path]
-    ],
-) -> list[pathlib.Path]:
-    (
-        username,
+def _original_files_file_download(
```
Comment (uriii3) [inline copernicusmarine/download_functions/download_original_files.py:514]: weird name for the function, no? or do you refer to all the original files from the file requested?

File: pyproject.toml [hunk 4]
```diff
@@ -20,7 +19,6 @@ dask = ">=2022"
 netCDF4 = ">=1.5.4"
 boto3 = ">=1.25"
 semver = ">=3.0.2"
-nest-asyncio = ">=1.5.8"
```
Comment (uriii3) [inline pyproject.toml:23]: hehe <3

**Comment by uriii3** [comment]
have you noticed or check the behaviour time wise? does it grow the time used or is it more or less the similar? maybe even less? if it increases, is it worth to be able to get rid of nest-asyncio?

File: copernicusmarine/catalogue_parser/catalogue_parser.py [hunk 5]
```diff
@@ -49,87 +41,63 @@
...
         ) as response:
-            return await response.json()
+            return response.json()
 
-    async def close(self) -> None:
-        await self.session.close()
+    def __enter__(self):
+        return self
 
-    def _retry_policy(self, info: RetryInfo) -> RetryPolicyStrategy:
```
Comment (uriii3) [inline copernicusmarine/catalogue_parser/catalogue_parser.py:76]: are we getting rid of the retries or is it handled internally somewhere else? the errors seem to be handle another place, but does this mean that we have less control on the behaviour of them?

Comment (renaudjester) [reply copernicusmarine/catalogue_parser/catalogue_parser.py:76]: We are using the `requests` error retry directly 

Yeah I guess a bit less control 🤔

**Comment by renaudjester** [comment]
> have you noticed or check the behaviour time wise? does it grow the time used or is it more or less the similar? maybe even less? if it increases, is it worth to be able to get rid of nest-asyncio?

Yeah I have run some tests and it was very similar times :D

File: copernicusmarine/download_functions/download_original_files.py (hunk 2 above)
Comment (renaudjester) [reply copernicusmarine/download_functions/download_original_files.py:261]: Yeah I like it better I think it's easier than importing the typing hint

File: copernicusmarine/catalogue_parser/catalogue_parser.py (hunk 5 above)
Comment (uriii3) [reply copernicusmarine/catalogue_parser/catalogue_parser.py:76]: Well, not that I like it ee, I feel like it's easier then, so good!

**Pull request merged**